plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/HiveHadoop.java

```diff
@@ -77,12 +95,23 @@ protected void setupContainer()
     public void start()
     {
         super.start();
+
+        // Even though Hadoop is accessed by proxy, Hadoop still tries to resolve hadoop-master
+        // (e.g: in: NameNodeProxies.createProxy)
+        // This adds a static resolution for hadoop-master to docker container internal ip
+        NetUtils.addStaticResolution(HOST_NAME, getContainerIpAddress());
+
         log.info("Hive container started with addresses for metastore: %s", getHiveMetastoreEndpoint());
     }
 
     public HostAndPort getHiveMetastoreEndpoint()
     {
-        return getMappedHostAndPortForExposedPort(HIVE_METASTORE_PORT);
+        return HostAndPort.fromParts(hostAddress, getMappedPortForExposedPort(HIVE_METASTORE_PORT));
```

Comment:
The previous version (`getMappedHostAndPortForExposedPort(HIVE_METASTORE_PORT)`) looked better.

Reply:
do you know why?

cc @wendigo @losipiuk